jscomp/core/lam_compile_primitive.ml

```diff
@@ -416,7 +416,7 @@ let translate output_prefix loc (cxt : Lam_compile_context.t)
       | _ -> assert false)
   | Pbigintmax -> (
       match args with
-      | [ { expression_desc = Number (Float _) } as a; { expression_desc = Number (Float _) } as b ]
+      | [ { expression_desc = Number (BigInt _) } as a; { expression_desc = Number (BigInt _) } as b ]
         when Js_analyzer.is_okay_to_duplicate a && Js_analyzer.is_okay_to_duplicate b ->
             E.econd (E.js_comp Cgt a b) a b
```

Comment:
Use `E.bigint_comp` here should improve the outputs

Reply:
Got it, thanks. https://github.com/rescript-lang/rescript-compiler/pull/7088/commits/77a46307c78cf832f2cb5d127846c7d8a4d7b63d

Reply:
Hmm ok `E.econd` itself doesn't seem to be optimized enough yet